r/src/compute.cpp

```diff
@@ -233,6 +233,33 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
                                      max_replacements);
   }
 
+  if (func_name == "split_pattern") {
+    using Options = arrow::compute::SplitPatternOptions;
+    int64_t max_splits = -1;
```

Comment:
Rather than setting defaults manually, can you do `auto out = std::make_shared<Options>(Options::Defaults());` and then just set them if they're provided? (We do this above for other types.)

Reply:
IIRC this didn't work when I tried it

Reply:
They probably should be added in the C++ library, can you do that and/or make another JIRA?

Reply:
@thisisnic could you make a Jira for this please? Thanks!

Reply:
Done

Reply:
Thinking about this, @nealrichardson and @ianmcook , doesn't it make sense for it not to have defaults, given that the `pattern` is one of the options, and it might not make sense to pre-specify a pattern to split the string on?  Neither of the R functions `strsplit` nor `stringr::str_split` have defaults.  

Could one perhaps make a case for `pattern` moving from being an option to instead being a second input, similarly to how `arrow::compute::filter` and `arrow::compute::take` operate?

Reply:
That all makes good sense to me but I'm not up to speed on what design conventions we follow in the C++ compute functions. Maybe it'd be best to start a Zulip thread to ask folks about this (Joris, Antonie, Maarten Breddels) to get a better idea of the best change to suggest (if any).